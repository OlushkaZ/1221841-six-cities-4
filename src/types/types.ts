export interface Offer {
  pictures: Array<string>,
  previewImage: string,
  title: string,
  description: Array<string>,
  premium: boolean,
  favourite: boolean,
  type: string,
  rating: number,
  bedrooms: number,
  guests: number,
  cost: number,
  conveniences: Array<string>,
  coordinates: Array<number>,
  owner: {
    avatar: string,
    name: string,
    pro: boolean,
    id: number,
    },
  id: number,
};

export interface CityCoordinates {
  coordinates: Array<number>,
  zoom: number,
}

export interface CityOffers {
  city: string,
  cityCoordinates: CityCoordinates,
  offers: Array<Offer>,
}

export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH =`NO_AUTH`,
}

export interface Classes {
  card: string,
  wrapper: string,
  cards?: string,
  map?: string,
  info?: string,
}

export interface Review {
  comment: string,
  date: string,
  rating: number,
  id: number,
  user: {
    avatar: string,
    name: string,
    isPro: boolean,
    userId: number,
  }
}


export enum Sorting {
  DEFAULT = `popular`,
  TO_HIGHT = `to-high`,
  TO_LAW = `to-low`,
  TOP_RATED = `top-rated`,
}
