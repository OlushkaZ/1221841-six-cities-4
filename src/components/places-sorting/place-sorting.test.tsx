import * as React from "react";
import * as renderer from "react-test-renderer";
import PlacesSorting from "./places-sorting";
import {Sorting} from "../../types/types";

const props = {
  activeClass: `places__options--opened`,
  onSortingListItemClick: jest.fn(),
  activeSortingType: Sorting.DEFAULT,
  handleChangeToggleClick: jest.fn(),
};

it(`Render PlaceSorting`, () => {


  const tree = renderer
    .create(
        <PlacesSorting {...props}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
