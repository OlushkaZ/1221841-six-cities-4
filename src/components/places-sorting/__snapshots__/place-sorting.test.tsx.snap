// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PlaceSorting 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex={0}
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex={0}
      value="popular"
    >
      Popular
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
      tabIndex={0}
      value="to-high"
    >
      Price: low to high
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
      tabIndex={0}
      value="to-low"
    >
      Price: high to low
    </li>
    <li
      className="places__option false"
      onClick={[Function]}
      tabIndex={0}
      value="top-rated"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
